doc/content/xapi/internals/certificates.md

```diff
@@ -0,0 +1,105 @@
+
++++
+title = "Certificates and PEM Files"
++++
+
+Xapi uses certificates for secure communication within a pool and with
+external clients. These certificates are using the PEM file format and
+reside in the Dom0 file system. This documents explains the purpose of
+these files.
+
+##  Design Documents
+
+* [Design](../../design/user-certificates.md)
+
+## Paths
+
+Below are paths used by Xapi for certificates; additional certficates
+may be installed but they are not fundamental for Xapi's operation.
+
+```
+/etc/xensource/xapi-ssl.pem
+/etc/xensource/xapi-pool-tls.pem
+/etc/stunnel/certs-pool/1c111a1f-412e-47c0-9003-60789b839bc3.pem
+/etc/stunnel/certs-pool/960abfff-6017-4d97-bd56-0a8f1a43e51a.pem
+/etc/stunnel/xapi-stunnel-ca-bundle.pem
+/etc/stunnel/certs/
+/etc/stunnel/xapi-pool-ca-bundle.pem
+```
+
+
+## Fundamental Certificates
+
+Certificates that identify a host. These certificates are comprised of
+both a private and a public key. The public key may be distributed to
+other hosts.
+
+### xapi-ssl.pem
+
+This certificate identifies a host for extra-pool clients.
+
+This is the certificate used by the API HTTPS server that clients like
+XenCenter or CVAD connect to. On installation of XenServer it is auto
+generated but can be updated by a user using the API. This is the most
+important certificate for a user to establish an HTTPS connection to a
+pool or host to be used as an API.
+
+* /etc/xensource/xapi-ssl.pem
+* contains private and public key for this host
+* `Host.get_server_certificate` API call
+* referenced by /etc/stunnel/xapi.conf
+* `xe host-server-certificate-install` XE command to replace the
+  certificate.
+* See below for xapi-stunnel-ca-bundle for additional certificates that
```

Comment:
It's worth mentioning that `reset_server_certificate` generates a new, self-signed one